relayer/src/cache.rs

```diff
@@ -84,20 +94,20 @@ impl Cache {
         &self,
         id: &PortChannelId,
         f: F,
-    ) -> Result<ChannelEnd, E>
+    ) -> CacheResult<ChannelEnd, E>
     where
         F: FnOnce() -> Result<ChannelEnd, E>,
     {
         if let Some(chan) = self.channels.get(id) {
             // If cache hit, return it.
-            Ok(chan)
+            Ok((chan, CacheStatus::Hit))
         } else {
             // Only cache a channel end if the channel is open.
             let chan = f()?;
             if chan.state().is_open() {
                 self.channels.insert(id.clone(), chan.clone());
             }
-            Ok(chan)
+            Ok((chan, CacheStatus::Miss))
```

Comment:
Hmm I imagine it might be cleaner to add calls to `telemetry!` directly in the cache layer (in these methods here), instead of doing it at the caller (in the `relayer/src/chain/handle/cache.rs`). What do you think?

Reply:
In the cache layer we don't know what is the exact type of query that has been performed, hence why I chose to add this code here. Also, it keeps a nice separation between the cache and the queries+metrics.